### PR TITLE
Withdraw repay borrow handling

### PR DESCRIPTION
**Pull request by dmytro-shash**

### CODE REVIEW

File: contracts/dtoken/src/common.rs [hunk 1]
```diff
@@ -212,6 +212,10 @@ impl Contract {
     pub fn increase_contract_balance(&mut self, amount: WBalance) -> Balance {
         self.set_contract_balance(self.get_contract_balance() + Balance::from(amount))
     }
+
+    pub fn decrease_contract_balance(&mut self, amount: WBalance) -> Balance {
```
Comment (TymofiiSavinov) [inline contracts/dtoken/src/common.rs:216]: There may be a negative value here?

File: contracts/dtoken/tests/sim/liquidation/liquidation_fail_as_liquidator_is_borrower.rs [hunk 2]
```diff
@@ -1,183 +1,182 @@
...
-    let user_balance: Balance = view_balance(
-        &controller,
-        Supply,
-        liquidator.account_id(),
-        wnear_market.account_id(),
-    );
-
-    assert_eq!(user_balance, 0, "Supply balance on dtoken should be 0");
-}
+// use crate::utils::{add_market, borrow, initialize_controller, initialize_two_dtokens, initialize_two_utokens, liquidate, mint_and_reserve, mint_tokens, new_user, set_price, supply, view_balance};
```
Comment (TymofiiSavinov) [inline contracts/dtoken/tests/sim/liquidation/liquidation_fail_as_liquidator_is_borrower.rs:1]: Why everything is commented out?

File: contracts/dtoken/tests/sim/liquidation/liquidation_fail_not_allowed_amount.rs [hunk 3]
```diff
@@ -1,182 +1,182 @@
...
-    let user_balance: Balance = view_balance(
-        &controller,
-        Supply,
-        liquidator.account_id(),
-        wnear_market.account_id(),
-    );
-
-    assert_eq!(user_balance, 0, "Supply balance on dtoken should be 0");
-}
+// use crate::utils::{
```
Comment (TymofiiSavinov) [inline contracts/dtoken/tests/sim/liquidation/liquidation_fail_not_allowed_amount.rs:1]: same?

File: contracts/dtoken/src/common.rs (hunk 1 above)
Comment (dmytro-shash) [reply contracts/dtoken/src/common.rs:216]: we've solved this issue by having Reserve action, which aims not to drain the contract's balance. 

yet this is a fair comment, I will add that assert there

**Review by TymofiiSavinov — APPROVED** [review]

**Review by TymofiiSavinov — APPROVED** [review]

**Pull request merged**